### PR TITLE
NiTs update

### PR DESCRIPTION
**Pull request by gorryfair**

### CODE REVIEW

File: draft-ietf-quic-ack-frequency.md [hunk 1]
```diff
@@ -418,11 +417,11 @@ the network, and will suffer degraded performance if acknowledgments are delayed
 excessively.
 
 To enable a sender to respond to potential network congestion, a sender SHOULD
-cause a receiver to send an acknowledgement at least once per RTT if there are
+cause a receiver to send an acknowledgement at least once per round-trip time (RTT) if there are
 unacknowledged ack-eliciting packets in flight. A sender can accomplish this by
-sending an IMMEDIATE_ACK frame once per round-trip time (RTT), or it can set the
+sending an IMMEDIATE_ACK frame once per RTT, or it can set the
 Ack-Eliciting Threshold and Request Max Ack Delay values to be no more than a
```
Comment (ana-cc) [inline draft-ietf-quic-ack-frequency.md:423]: maybe 'no more' should be 'no larger'?

File: draft-ietf-quic-ack-frequency.md [hunk 2]
```diff
@@ -452,11 +451,11 @@ rely on an accurate RTT estimate, such as time-threshold loss detection (Section
...
-To limit these consequences of reduced acknowledgement frequency, a sender
-SHOULD cause a receiver to send an acknowledgement at least once per RTT if
+To limit the consequences of reduced acknowledgement frequency, a sender
+SHOULD cause a receiver to send an acknowledgement at least once per round-trip time
+(RTT) if
 there are unacknowledged ack-eliciting packets in flight. A sender can
-accomplish this by sending an IMMEDIATE_ACK frame once per round-trip time
-(RTT), or it can set the Ack-Eliciting Threshold and Request Max Ack Delay
+accomplish this by sending an IMMEDIATE_ACK frame once per RTT, or it can set the Ack-Eliciting Threshold and Request Max Ack Delay
 values to be no more than a congestion window and an estimated RTT,
```
Comment (ana-cc) [inline draft-ietf-quic-ack-frequency.md:459]: maybe 'no more' should be 'no larger'?

... this feels a bit repetitive

File: draft-ietf-quic-ack-frequency.md [hunk 3]
```diff
@@ -479,8 +478,14 @@ new ACK_FREQUENCY frame immediately upon confirmation of connection migration.
 
 
 # Security Considerations
+
 TBD.
 
+QUIC ACK_FREQUENCY frames are protected by the normal quic security mechanisms {{QUIC-TRANSPORT}}).
```
Comment (ana-cc) [inline draft-ietf-quic-ack-frequency.md:484]: s/quic/QUIC

File: draft-ietf-quic-ack-frequency.md [hunk 4]
```diff
@@ -479,8 +478,14 @@ new ACK_FREQUENCY frame immediately upon confirmation of connection migration.
...
 
 # Security Considerations
+
 TBD.
 
+QUIC ACK_FREQUENCY frames are protected by the normal quic security mechanisms {{QUIC-TRANSPORT}}). 
+Potentially an ACK_FREQUENCY frame could allow a DOS attack on the return path of a standards-compliant 
+receiver by a malicious/poorly configured server, however the sender has control of many aspects of 
+the receiver behaviour and this as a threat from a well-configured sender. 
+
```
Comment (ana-cc) [inline draft-ietf-quic-ack-frequency.md:488]: hmm, the first sentence can be rewritten as:

Improperly configured  or malicious senders could cause standards-compliant receivers to acknowledge more frequently than their available resources permit, thus causing a Denial Of Service (DOS) attack.  

I'm not sure about the second sentence, what was the gist of it? Senders have control of many aspects of the receiver behaviour anyway and so this is not an issue for well-configured servers?

**Comment by gorryfair** [comment]
Seems like the new text updates in this PR will resolve these problems.

File: draft-ietf-quic-ack-frequency.md [hunk 5]
```diff
@@ -354,12 +354,11 @@ acknowledgements.
...
 An endpoint SHOULD send an immediate acknowledgement when a packet marked
 with the ECN Congestion Experienced (CE) codepoint in the IP header is
-received and the previously received packet was not marked CE.
-
+received and the previously received packet was not marked CE {{?RFC3168}}.
 Doing this maintains the peer's response time to congestion events, while also
 reducing the ACK rate compared to {{Section 13.2.1 of QUIC-TRANSPORT}} during
 extreme congestion or when peers are using DCTCP {{?RFC8257}} or other
-congestion controllers that mark more frequently than classic ECN {{?RFC3168}}.
+congestion controllers (e.g. {{I-D.ietf-tsvwg-aqm-dualq-coupled}} that mark more frequently than classic ECN {{?RFC3168}}.
```
Comment (ana-cc) [inline draft-ietf-quic-ack-frequency.md:361]: (e.g. {{I-D.ietf-tsvwg-aqm-dualq-coupled}} needs a ')' at the end

File: draft-ietf-quic-ack-frequency.md [hunk 6]
```diff
@@ -452,13 +451,14 @@ rely on an accurate RTT estimate, such as time-threshold loss detection (Section
...
+SHOULD cause a receiver to send an acknowledgement at least once per round-trip time
+(RTT) if
 there are unacknowledged ack-eliciting packets in flight. A sender can
-accomplish this by sending an IMMEDIATE_ACK frame once per round-trip time
-(RTT), or it can set the Ack-Eliciting Threshold and Request Max Ack Delay
-values to be no more than a congestion window and an estimated RTT,
-respectively.
+accomplish this by sending an IMMEDIATE_ACK frame once per RTT, or it can set the 
+Ack-Eliciting Threshold and Request Max Ack Delay
+values to be respectively smaller than a congestion window and nolonger than
```
Comment (ana-cc) [inline draft-ietf-quic-ack-frequency.md:460]: s/nolonger/no longer/

File: draft-ietf-quic-ack-frequency.md [hunk 7]
```diff
@@ -479,8 +479,14 @@ new ACK_FREQUENCY frame immediately upon confirmation of connection migration.
 
 
 # Security Considerations
+
 TBD.
 
+AN Improperly configured or malicious sender could cause standards-compliant receivers to
```
Comment (ana-cc) [inline draft-ietf-quic-ack-frequency.md:485]: s/AN Improperly/An improperly/

File: draft-ietf-quic-ack-frequency.md [hunk 8]
```diff
@@ -479,8 +479,14 @@ new ACK_FREQUENCY frame immediately upon confirmation of connection migration.
 
 
 # Security Considerations
+
 TBD.
 
+AN Improperly configured or malicious sender could cause standards-compliant receivers to 
+acknowledge more frequently than their available resources permit, thus causing a 
+Denial Of Service (DOS) attack. However, A sender has control of many aspects of
```
Comment (ana-cc) [inline draft-ietf-quic-ack-frequency.md:487]: s/A/a/

File: draft-ietf-quic-ack-frequency.md [hunk 9]
```diff
@@ -479,8 +479,14 @@ new ACK_FREQUENCY frame immediately upon confirmation of connection migration.
 
 
 # Security Considerations
+
 TBD.
 
+AN Improperly configured or malicious sender could cause standards-compliant receivers to 
+acknowledge more frequently than their available resources permit, thus causing a 
+Denial Of Service (DOS) attack. However, A sender has control of many aspects of 
+the receiver behaviour and therefore there is no additional threat from a well-configured sender.
```
Comment (ana-cc) [inline draft-ietf-quic-ack-frequency.md:488]: s/there is no additional threat from a well-configured sender/for a well-configured sender this causes no additional threat/

**Review by ana-cc — CHANGES_REQUESTED** [review]

**Pull request merged**